saleor/payment/gateways/adyen/webhooks.py

```diff
@@ -486,3 +488,31 @@ def handle_webhook(request: WSGIRequest, gateway_config: "GatewayConfig"):
         event_handler(notification, gateway_config)
         return HttpResponse("[accepted]")
     return HttpResponseNotFound()
+
+
+def handle_additional_actions(request: WSGIRequest, payment_details: Callable):
+    payment_id = request.GET["payment"]
+    _type, payment_pk = from_global_id(payment_id)
+    try:
+        payment = Payment.objects.get(pk=payment_pk)
+    except ObjectDoesNotExist:
+        raise Exception("Cannot perform payment. Payment does not exists.")
+
+    data = json.loads(payment.extra_data)
+    return_url = payment.return_url
+
+    if not return_url:
+        raise Exception("Cannot perform payment. Lack of data about returnUrl.")
```

Comment:
Not sure if low lvl `Exception` is a good way.  The payment module has its own error lvl `PaymentError`.
Also. Is it expected that we don't catch these raises? Won't it close the python proces?

Reply:
I am returning `HttpResponseNotFound` right now.